pkg/traces/exporter/client_authenticator.go

```diff
@@ -0,0 +1,30 @@
+package exporter
+
+import (
+	"context"
+	"fmt"
+)
+
+// Implements google.golang.org/grpc/credentials.PerRPCCredentials interface
+type clientAuthenticator struct {
+	token string
+}
```

Comment:
This makes sense to me, but I think it would benefit from the addition of `SetNewToken(token string)`

That would allow the token to be updated, without create a new exporter.

Reply:
I suspect it would be super clean to create one, and register it with `knapsack.RegisterChangeObserver`

Maybe pull that into `New(k knapsack)`? Not sure...

Reply:
Yes, definitely cleaner! I'll update

Reply:
This is updated -- I ended up making `TraceExporter` subscribe to the `ingest` updates, and swap in the token when pinged.